lib/async.array.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const { promisify } = require('util');
+
+const timeout = promisify(resolve => setTimeout(resolve, 0));
+
+class AsyncArray extends Array {
+  constructor(...args) {
+    super(...args);
+
+    this.min = 5;
+    this.ratio = 1.5;
+  }
+
+  set percent(value) {
+    if (value < 1 && value >= 0) {
+      this.ratio = value / (1 - value);
+    } else {
+      throw new Error('The percen must be less than 1 and greater ' +
```

Comment:
```suggestion
      throw new Error('The percent must be less than 1 and greater ' +
```

Reply:
This should be RangeError, also I'd use `...must be >= 0 and < 1` or even `...must be in range [0, 1)`.